internal/metrics/otelmetrics/factory_test.go

```diff
@@ -40,10 +40,34 @@ func findMetric(t *testing.T, registry *promReg.Registry, name string) *promMode
 			return mf
 		}
 	}
-	require.NotNil(t, name, "Expected to find Metric Family")
+	require.Fail(t, "Expected to find Metric Family")
 	return nil
 }
 
+func checkLabels(t *testing.T, expectedLabels map[string]string, labels []*promModel.LabelPair) {
+	for key, value := range expectedLabels {
+		found := false
+		for _, label := range labels {
+			if label.GetName() == key {
+				assert.Equal(t, value, label.GetValue())
+				found = true
+				break
+			}
+		}
+		assert.True(t, found, "Label %s not found", key)
```

Comment:
considering that expectedLabels is a map, I would recommend instead to convert `labels` also to a map and then simply call `assert.Equals`, which will print a nice diff (if there is one). In your version all you get is uninformative true/false.

Reply:
and then you can call assert directly from the test functions, not here, e.g.
```
assert.Equal(t, expectedLabels, promLabelsToMap(...))
```